### PR TITLE
Add a recipe for bytecode package

### PR DESCRIPTION
**Pull request by MatthieuDartiailh**
Bytecode is Python bytecode assembler disassembler created initially by vstinner. It will be used in the next version of enaml for which a feedstock does exist. 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/bytecode/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: f78a312880173f1d20aef2eceb8eb959dc6c2167966b757fef201b8af839f802
+
+build:
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  skip: True  # [py<35]
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (npavlovikj) [inline recipes/bytecode/meta.yaml]: I think this recipe can be `noarch`, https://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages. Thus, please remove the `skip` line and add `noarch: python` instead.

File: recipes/bytecode/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+requirements:
+  host:
+    - python
```
Comment (npavlovikj) [inline recipes/bytecode/meta.yaml]: Use `python >=3.5`.

File: recipes/bytecode/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    - pip
+  run:
+    - python
```
Comment (npavlovikj) [inline recipes/bytecode/meta.yaml]: Use `python >=3.5`.

File: recipes/bytecode/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    - aenum  # [py<36]
```
Comment (npavlovikj) [inline recipes/bytecode/meta.yaml]: Please try removing the selector, `# [py<36]`.

**Comment by npavlovikj** [comment]
@MatthieuDartiailh , the AppVeyor error can be ignored. I added few comments, so please address them. Moreover, please clean the recipe and remove all comments and extra line in the `about` section.

File: recipes/bytecode/meta.yaml (hunk 4 above)
Comment (MatthieuDartiailh) [reply recipes/bytecode/meta.yaml]: aenum is not required on python 3.6 and above which is why I used the selector. Should we install it unconditionally even if it is not used ?

**Comment by MatthieuDartiailh** [comment]
@npavlovikj I addressed all your comments even though I am not convinced this package should be noarch as explained above. Thanks for the quick review BTW.

File: recipes/bytecode/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: f78a312880173f1d20aef2eceb8eb959dc6c2167966b757fef201b8af839f802
+
+build:
+  noarch: True
```
Comment (npavlovikj) [inline recipes/bytecode/meta.yaml]: This should be `noarch: python`.

File: recipes/bytecode/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  license: MIT
+  license_family: MIT
+  license_file: COPYING
+  summary: 'A Python module to generate and modify Python bytecode.'
+  doc_url: https://bytecode.readthedocs.io/
+  dev_url: https://github.com/vstinner/bytecode
+
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
```
Comment (npavlovikj) [inline recipes/bytecode/meta.yaml]: Please remove these comments as well.

File: recipes/bytecode/meta.yaml (hunk 4 above)
Comment (npavlovikj) [reply recipes/bytecode/meta.yaml]: Ah, I see. Yeah, if it is not needed at all, you can remove it. I think CF builds against Python >=3.6 anyway.

Comment (MatthieuDartiailh) [reply recipes/bytecode/meta.yaml]: I removed aeunm and in consequence bumped the minimum python version to 3.6. I could not confirm that CF only build on python 3.6 and above so it sounds safer to make sure we do not generate a bad package.

**Comment by MatthieuDartiailh** [comment]
ping @npavlovikj

**Comment by npavlovikj** [comment]
The recipe looks good @MatthieuDartiailh , thank you for contributing 🎉

**Comment by npavlovikj** [comment]
> @npavlovikj I addressed all your comments even though I am not convinced this package should be noarch as explained above. Thanks for the quick review BTW.

In general, if a recipe doesn't require compilers or architecture dependent packages, it can be `noarch`, https://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages.

**Comment by MatthieuDartiailh** [comment]
Thanks @npavlovikj

**Pull request merged**